tests/testapp/tests/test_api.py

```diff
@@ -747,7 +747,7 @@ def test_push_fails_for_pull_transfersession(self):
         rec_1 = self.build_buffer_item(push=False, filter=self.default_push_filter)
         rec_2 = self.build_buffer_item(transfer_session=rec_1.transfer_session)
         rec_3 = self.build_buffer_item(transfer_session=rec_1.transfer_session)
-        self.make_buffer_post_request([rec_1, rec_2, rec_3], expected_status=400)
+        self.make_buffer_post_request([rec_1, rec_2, rec_3], expected_status=403)
```

Comment:
This test was actually failing based on non valid sub partitions and not on push/pull transfer session mismatch. It now fails in the correct spot and returns the correct error code.